viewing_party/party.py

```diff
@@ -1,23 +1,116 @@
+import statistics
+from statistics import mode
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if None in (title, genre, rating):
+        return None
+    else:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title != movie["title"]:
+            continue
+        elif movie["title"] == title:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+            return user_data
+        
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    if not len(user_data["watched"]):
+        return 0.0
+    ratings = [movie["rating"] for movie in user_data["watched"]]
+    return statistics.mean(ratings)
+
+def get_most_watched_genre(user_data):
+    if not len(user_data["watched"]):
+        return None
+    genres = [movie["genre"]for movie in user_data["watched"]]
+    return mode(genres)
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    user_movies = get_user_movies(user_data)
+    friend_movies = get_friends_movies(user_data)

```

Comment:
Prefer variable names to include datatype when possible, especially for projects like this where we're working with several different datatypes because it makes your code more readable.

`user_movies_list` and `friend_movies_list`